harness/determined/estimator/_estimator_context.py

```diff
@@ -142,3 +144,26 @@ def allgather_metrics(self, metrics: Any) -> List:
         if self._allgather_fn is None:
             raise AssertionError("allgather_metrics must not be called before training begins")
         return self._allgather_fn(metrics)
+
+    def _build_allgather_op(self, f: Callable) -> Callable:
+        """Build an op that uses allgather in a way that is safely sequentialized."""
+
+        def decorated(*arg: Any, **kwarg: Any) -> tf.Operation:
```

Comment:
non-blocking: seems like we are wrapping only internal functions so shouldn't matter that much, but doesn't hurt to add in `functools.wraps()` here.

Reply:
good idea